src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatastore.scala

```diff
@@ -0,0 +1,231 @@
+package tech.cryptonomic.conseil.tezos
+
+import tech.cryptonomic.conseil.tezos.Tables.{BalanceUpdatesRow, OperationGroupsRow, OperationsRow}
+import tech.cryptonomic.conseil.tezos.TezosTypes.{
+  Account,
+  AccountId,
+  Block,
+  BlockReference,
+  BlockTagged,
+  Delegate,
+  PublicKeyHash,
+  Voting
+}
+import tech.cryptonomic.conseil.tezos.repositories.{
+  AccountsRepository,
+  BlocksRepository,
+  DelegatesRepository,
+  OperationsRepository,
+  VotingRepository
+}
+import tech.cryptonomic.conseil.util.Conversion
+
+import cats.implicits._
+import cats.effect._
+import cats.data.Kleisli
+import slick.dbio.DBIO
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.implicits._
+import scala.concurrent.ExecutionContext
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+  * Functions for writing Tezos data to a database.
+  */
+class TezosDatastore(
```

Comment:
Could you explain more explicitly how creating this data store helps with the forking work?

Reply:
I needed to have a clear picture of which operations were used to read or join from the block's table, and which were writing there.

That's because I couldn't easily figure out which operations would need to join with the invalidated tables for read, and which should not (e.g. some operations will need to read invalidated blocks too, to take decisions when downloading the forked blocks)